chall09/jbaringo.c

```diff
@@ -0,0 +1,194 @@
+#include <stdlib.h>
+#include <string.h>
+
+int         ft_isdigit(char c)
+{
+    if (c < '0' || c > '9')
+        return (0);
+    return (1);
+}
+
+
+int             get_size(unsigned long long num)
+{
+    int i = 0;
+
+    while (num >= 10)
+    {
+        i++;
+        num /= 10;
+    }
+    return (i + 1);
+}
+
+char			*ft_long_itoa(unsigned long long num)
+{
+	char		    *str;
+	int	index;
+	int	size;
+    
+	size = get_size(num);
+	index = 0;
+	if (!(str = malloc(sizeof(char) * (size + 1 ))))
+		return (0);
+
+	index = size - 1;
+	while (num >= 10)
+	{
+		str[index--] = (num % 10 + 48);
+		num /= 10;
+	}
+	str[index] = (num % 10 + 48);
+	str[size] = '\0';
+	return (str);
+}
+
+unsigned long long  ft_long_atoi(char *str)

```

Comment:
Just a hint, there was a special function to transform from `char *` to `unsigned long long` called `strtoull()` that could also help you with error handling, recoding your own version of a better `atoi()` was a bit of overkill, but good job nonetheless.

The problem here and in your error checking, is that you forgot to check if the provided `char *` was `> 10000000000000000000` and if it was superior to `ULLONG_MAX` (from `<limits.h>`), which `strtoull()` would have mostly handled for you.